video/out/wayland_common.c

```diff
@@ -2095,12 +2092,7 @@ static void set_window_bounds(struct vo_wayland_state *wl)
 
 static int spawn_cursor(struct vo_wayland_state *wl)
 {
-    struct vo_wayland_seat *s = wl->cursor_seat;
-    /* Don't use this if we have cursor-shape. */
-    if (s && s->cursor_shape_device)
-        return 0;
```

Comment:
I don't think you can remove this. The surface entrance event and the output done event both call this to potentially rescale the client side cursor.

Reply:
But you'll just be loading the cursor themes for no reason. If we have cursor shape, client side cursors should never be used.

Reply:
You don't have to deal with seats. You can just check if there's `cursor_shape_manager`.